### ISSUE TITLE
CMD+H / CMD+W shortcuts support on macOS

### ISSUE THREAD

**Issue by pieterclaerhout** [Microsoft/AzureStorageExplorer#99]
**Storage Explorer Version: 1.0**

**OS Version: any macOS version**

**Steps to Reproduce:**

1. Open Storage Explorer
2. Presss CMD+H on the keyboard.
3. Nothing happens

**Expected Experience:**

The expected behaviour is that the app gets hidden, just like all other mac apps are doing.

**Actual Experience:**

Nothing happens as the CMD+H shortcut is not defined in Storage Explorer as you can see in the following screenshots:

Azure Storage Explorer

<img width="301" alt="screen shot 2018-04-27 at 10 59 08" src="https://user-images.githubusercontent.com/74731/39354231-51a9f5b2-4a0a-11e8-9f3b-79a1e44383ef.png">

Standard Mac App (Safari in this case)

<img width="267" alt="screen shot 2018-04-27 at 10 59 01" src="https://user-images.githubusercontent.com/74731/39354244-5cd659f8-4a0a-11e8-92b4-30fe6eaa8856.png">

Azure Storage Explorer should have the standard options in the menu:

- Hide Storage Explorer
- Hide Others
- Show All

Also, the shortcut CMD+F4 is very not-mac. Closing a window on a mac is done by CMD+W, so you might want to change that as well.

Also, the commands `New Window` and `Close Window`are normally defined under the `File`menu on macOS.

**Comment by MRayermannMSFT** [comment]
Hey @pieterclaerhout , I'm working on this right now. Let me know if you have any other feedback regarding our menus on OSX, as I'm going to be taking a comprehensive look to see if what we're doing feels/looks right.

**Comment by pieterclaerhout** [comment]
New Window should actually be in the File menu as well.

And in the File menu, there should also be a "Close" menu item with the shortcut "CMD+W" which is used to close the current window.

"Close Editor" under the "Microsoft Azure Storage Explorer" is also having a very non-mac shortcut CMD+F4.

The standard behaviour on mac is to use CMD+W for both closing the tabs as well as the window. If a tab is open, CMD+W should close the tab. If no tabs are open, CMD+W should close the window…

**Comment by pieterclaerhout** [comment]
Also Help > Reload should use CMD+R as the shortcut on the mac instead of CTRL+R.

**Comment by MRayermannMSFT** [comment]
Alright, I was able to do almost all of your feedback.

Here's new app menu:
![image](https://user-images.githubusercontent.com/21112697/41004112-a390700e-68cd-11e8-9829-ab4ef69d01a6.png)
There is now "Hide Storage Explorer", "Hide Others" and "Show All".

Here's the File menu (which previously didn't show on macOS):
![image](https://user-images.githubusercontent.com/21112697/41004132-b21e9394-68cd-11e8-878e-998a428b55e5.png)
I moved "New Window" and "Close Current Editor" to this menu, and also added "Close Window'

And here's the Help menu:
![image](https://user-images.githubusercontent.com/21112697/41004135-b628a862-68cd-11e8-9ab6-44ac621ff08c.png)
With reload now having CMD+R.

One caveat to these changes though is that "Quit" and "Close Window" do the same thing. When you click "New Window" we aren't just opening a window. Instead, we are starting up an entirely new instance of Storage Explorer. Doing the right thing, just opening a window, is in our backlog, so when we get to that, we'll make the behavior of "Close Window" actually just close the current window.
